### ISSUE TITLE
warn on iterating over an `Option`

### ISSUE THREAD

**Issue by oli-obk** [Manishearth/rust-clippy#584]
related to #182, but #182 should take precedence over this check

suggest that 

```rust
for x in option {
    println!("{}", x);
}
```

should be 

```rust
if let Some(x) = option {
    println!("{}", x);
}

**Comment by llogiq** [comment]
One slight complication is the for loop expansion (this will look like a `loop` containing a `match` statement, the `ExpnInfo` from the `span` will have the relevant information). The `option` type check has been done before (`src/methods.rs` has a few instances), and the rest is fairly easy.

**Comment by devonhollowood** [comment]
I can do this.

**Comment by Manishearth** [comment]
Go ahead!

**Comment by mcarton** [comment]
This issue can be closed.

**Comment by Manishearth** [comment]
#591
